build-logic/convention/src/main/kotlin/AndroidApplicationJacocoConventionPlugin.kt

```diff
@@ -14,22 +14,21 @@
  *   limitations under the License.
  */
 
-import com.android.build.api.variant.ApplicationAndroidComponentsExtension
-import com.google.samples.apps.nowinandroid.configureJacoco
+import com.android.build.gradle.internal.dsl.BaseAppModuleExtension
 import org.gradle.api.Plugin
 import org.gradle.api.Project
 import org.gradle.kotlin.dsl.getByType
 
 class AndroidApplicationJacocoConventionPlugin : Plugin<Project> {
     override fun apply(target: Project) {
         with(target) {
-            with(pluginManager) {
-                apply("org.gradle.jacoco")
-                apply("com.android.application")
+            val androidExtension = extensions.getByType<BaseAppModuleExtension>()
+
+            androidExtension.buildTypes.forEach { buildType ->
+                buildType.enableAndroidTestCoverage = true
+                buildType.enableUnitTestCoverage = true
```

Comment:
Let's use `configureEach` instead to be applied on all `buildTypes`. `forEach` will only use the current buildTypes, not the ones being created "after".

```suggestion
            androidExtension.buildTypes.configureEach {
                enableAndroidTestCoverage = true
                enableUnitTestCoverage = true
```